pkg/healthcheck/healthcheck.go

```diff
@@ -198,6 +216,21 @@ func IsCategoryError(err error, categoryID CategoryID) bool {
 	return false
 }
 
+// CheckRunResult Represents the result of running an individual checker
+type CheckRunResult struct {
```

Comment:
in the long term, I really like this approach.  In fact, I even suggested something similar on another PR.  But to keep this diff size down, I don't think we actually need to do this here.  I think we can keep the "error" and "warning" checks separate.